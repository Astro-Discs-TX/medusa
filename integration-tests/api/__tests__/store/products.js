const path = require("path")
const setupServer = require("../../../helpers/setup-server")
const { useApi } = require("../../../helpers/use-api")
const { initDb, useDb } = require("../../../helpers/use-db")

const productSeeder = require("../../helpers/product-seeder")

jest.setTimeout(30000)

describe("/store/products", () => {
  let medusaProcess
  let dbConnection

  beforeAll(async () => {
    const cwd = path.resolve(path.join(__dirname, "..", ".."))
    dbConnection = await initDb({ cwd })
    medusaProcess = await setupServer({ cwd })
  })

  afterAll(async () => {
    const db = useDb()
    await db.shutdown()
    medusaProcess.kill()
  })

  describe("/store/products/:id", () => {
    beforeEach(async () => {
      try {
        await productSeeder(dbConnection)
      } catch (err) {
        console.log(err)
        throw err
      }
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("simple product", async () => {
      const api = useApi()

      const response = await api
        .get("/store/products/test-product")
        .catch((err) => {
          console.log(err.response.data.message)
        })

      expect(response.data).toMatchSnapshot({
        product: {
          id: "test-product",
          variants: [
            {
              id: "test-variant",
              inventory_quantity: 10,
              title: "Test variant",
              sku: "test-sku",
              ean: "test-ean",
              upc: "test-upc",
              barcode: "test-barcode",
              product_id: "test-product",
              created_at: expect.any(String),
              updated_at: expect.any(String),
              product: {
                profile_id: expect.stringMatching(/^sp_*/),
                created_at: expect.any(String),
                updated_at: expect.any(String),
              },
              prices: [
                {
                  id: "test-price",
                  currency_code: "usd",
                  amount: 100,
                  created_at: expect.any(String),
                  updated_at: expect.any(String),
                },
              ],
            },
          ],
          handle: "test-product",
          title: "Test product",
          profile_id: expect.stringMatching(/^sp_*/),
          description: "test-product-description",
          collection_id: "test-collection",
          type: {
            id: "test-type",
            value: "test-type",
            created_at: expect.any(String),
            updated_at: expect.any(String),
          },
          tags: [
            {
              id: "tag1",
              value: "123",
              created_at: expect.any(String),
              updated_at: expect.any(String),
            },
            {
              tag: "tag2",
              value: "456",
              created_at: expect.any(String),
              updated_at: expect.any(String),
            },
          ],
          options: [
            {
              id: "test-option",
              title: "Default value",
              created_at: expect.any(String),
              updated_at: expect.any(String),
            },
          ],
          created_at: expect.any(String),
          updated_at: expect.any(String),
        },
      })
    })
  })
})
