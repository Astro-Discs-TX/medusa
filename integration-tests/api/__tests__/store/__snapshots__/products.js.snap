// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/store/products /store/products/:id includes default relations 1`] = `
Object {
  "product": Object {
    "collection": Object {
      "created_at": Any<String>,
      "deleted_at": null,
      "handle": "test-collection",
      "id": "test-collection",
      "metadata": null,
      "title": "Test collection",
      "updated_at": Any<String>,
    },
    "collection_id": "test-collection",
    "created_at": Any<String>,
    "deleted_at": null,
    "description": "test-product-description",
    "discountable": true,
    "handle": "test-product",
    "height": null,
    "hs_code": null,
    "id": "test-product",
    "images": Array [
      Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": "test-image",
        "metadata": null,
        "updated_at": Any<String>,
        "url": "test-image.png",
      },
    ],
    "is_giftcard": false,
    "length": null,
    "material": null,
    "metadata": null,
    "mid_code": null,
    "options": Array [
      Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": "test-option",
        "metadata": null,
        "product_id": "test-product",
        "title": "Test option",
        "updated_at": Any<String>,
        "values": Array [
          Object {
            "created_at": Any<String>,
            "deleted_at": null,
            "id": "test-option-value_1",
            "metadata": null,
            "option_id": "test-option",
            "updated_at": Any<String>,
            "value": "test-option-value_1",
            "variant_id": "test-variant",
          },
          Object {
            "created_at": Any<String>,
            "deleted_at": null,
            "id": "test-option-value_2",
            "metadata": null,
            "option_id": "test-option",
            "updated_at": Any<String>,
            "value": "test-option-value_2",
            "variant_id": "test-variant",
          },
        ],
      },
    ],
    "origin_country": null,
    "profile_id": StringMatching /\\^sp_\\*/,
    "subtitle": null,
    "tags": Array [
      Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": "tag1",
        "metadata": null,
        "updated_at": Any<String>,
        "value": "123",
      },
    ],
    "thumbnail": null,
    "title": "Test product",
    "type": Object {
      "created_at": Any<String>,
      "deleted_at": null,
      "id": "test-type",
      "metadata": null,
      "updated_at": Any<String>,
      "value": "test-type",
    },
    "type_id": "test-type",
    "updated_at": Any<String>,
    "variants": Array [
      Object {
        "allow_backorder": false,
        "barcode": "test-barcode",
        "created_at": Any<String>,
        "deleted_at": null,
        "ean": "test-ean",
        "height": null,
        "hs_code": null,
        "id": "test-variant",
        "inventory_quantity": 10,
        "length": null,
        "manage_inventory": true,
        "material": null,
        "metadata": null,
        "mid_code": null,
        "origin_country": null,
        "prices": Array [
          Object {
            "amount": 100,
            "created_at": Any<String>,
            "currency_code": "usd",
            "deleted_at": null,
            "id": "test-money-amount",
            "region_id": null,
            "sale_amount": null,
            "updated_at": Any<String>,
            "variant_id": "test-variant",
          },
        ],
        "product": Object {
          "collection_id": "test-collection",
          "created_at": Any<String>,
          "deleted_at": null,
          "description": "test-product-description",
          "discountable": true,
          "handle": "test-product",
          "height": null,
          "hs_code": null,
          "id": "test-product",
          "is_giftcard": false,
          "length": null,
          "material": null,
          "metadata": null,
          "mid_code": null,
          "origin_country": null,
          "profile_id": StringMatching /\\^sp_\\*/,
          "subtitle": null,
          "thumbnail": null,
          "title": "Test product",
          "type_id": "test-type",
          "updated_at": Any<String>,
          "weight": null,
          "width": null,
        },
        "product_id": "test-product",
        "sku": "test-sku",
        "title": "Test variant",
        "upc": "test-upc",
        "updated_at": Any<String>,
        "weight": null,
        "width": null,
      },
    ],
    "weight": null,
    "width": null,
  },
}
`;

exports[`/store/products List products includes default relations 1`] = `
Object {
  "count": 1,
  "limit": 100,
  "offset": 0,
  "products": Array [
    Object {
      "collection": Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "handle": "test-collection",
        "id": "test-collection",
        "metadata": null,
        "title": "Test collection",
        "updated_at": Any<String>,
      },
      "collection_id": "test-collection",
      "created_at": Any<String>,
      "deleted_at": null,
      "description": "test-product-description",
      "discountable": true,
      "handle": "test-product",
      "height": null,
      "hs_code": null,
      "id": "test-product",
      "images": Array [
        Object {
          "created_at": Any<String>,
          "deleted_at": null,
          "id": "test-image",
          "metadata": null,
          "updated_at": Any<String>,
          "url": "test-image.png",
        },
      ],
      "is_giftcard": false,
      "length": null,
      "material": null,
      "metadata": null,
      "mid_code": null,
      "options": Array [
        Object {
          "created_at": Any<String>,
          "deleted_at": null,
          "id": "test-option",
          "metadata": null,
          "product_id": "test-product",
          "title": "Test option",
          "updated_at": Any<String>,
          "values": Array [
            Object {
              "created_at": Any<String>,
              "deleted_at": null,
              "id": "test-option-value_1",
              "metadata": null,
              "option_id": "test-option",
              "updated_at": Any<String>,
              "value": "test-option-value_1",
              "variant_id": "test-variant",
            },
            Object {
              "created_at": Any<String>,
              "deleted_at": null,
              "id": "test-option-value_2",
              "metadata": null,
              "option_id": "test-option",
              "updated_at": Any<String>,
              "value": "test-option-value_2",
              "variant_id": "test-variant",
            },
          ],
        },
      ],
      "origin_country": null,
      "profile_id": StringMatching /\\^sp_\\*/,
      "subtitle": null,
      "tags": Array [
        Object {
          "created_at": Any<String>,
          "deleted_at": null,
          "id": "tag1",
          "metadata": null,
          "updated_at": Any<String>,
          "value": "123",
        },
      ],
      "thumbnail": null,
      "title": "Test product",
      "type": Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": "test-type",
        "metadata": null,
        "updated_at": Any<String>,
        "value": "test-type",
      },
      "type_id": "test-type",
      "updated_at": Any<String>,
      "variants": Array [
        Object {
          "allow_backorder": false,
          "barcode": "test-barcode",
          "created_at": Any<String>,
          "deleted_at": null,
          "ean": "test-ean",
          "height": null,
          "hs_code": null,
          "id": "test-variant",
          "inventory_quantity": 10,
          "length": null,
          "manage_inventory": true,
          "material": null,
          "metadata": null,
          "mid_code": null,
          "origin_country": null,
          "prices": Array [
            Object {
              "amount": 100,
              "created_at": Any<String>,
              "currency_code": "usd",
              "deleted_at": null,
              "id": "test-money-amount",
              "region_id": null,
              "sale_amount": null,
              "updated_at": Any<String>,
              "variant_id": "test-variant",
            },
          ],
          "product": Object {
            "collection_id": "test-collection",
            "created_at": Any<String>,
            "deleted_at": null,
            "description": "test-product-description",
            "discountable": true,
            "handle": "test-product",
            "height": null,
            "hs_code": null,
            "id": "test-product",
            "is_giftcard": false,
            "length": null,
            "material": null,
            "metadata": null,
            "mid_code": null,
            "origin_country": null,
            "profile_id": StringMatching /\\^sp_\\*/,
            "subtitle": null,
            "thumbnail": null,
            "title": "Test product",
            "type_id": "test-type",
            "updated_at": Any<String>,
            "weight": null,
            "width": null,
          },
          "product_id": "test-product",
          "sku": "test-sku",
          "title": "Test variant",
          "upc": "test-upc",
          "updated_at": Any<String>,
          "weight": null,
          "width": null,
        },
      ],
      "weight": null,
      "width": null,
    },
  ],
}
`;
